### ISSUE TITLE
Error when trying npm start

### ISSUE THREAD

**Issue by memento** [TwanoO67/bootstraping-ngx-admin-lte#79]
Hi thanks for your work.
I have the following error with your project. I wonder if you had to face it : 

```
USER@Dell:~/Bureau/labo/4/ng2-admin-lte$ npm install
```
```
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm WARN angular2-toaster@1.1.0 requires a peer of rxjs@5.0.0-beta.12 but none was installed.
```
```
USER@Dell:~/Bureau/labo/4/ng2-admin-lte$ npm start
```
```
> ng2-admin-lte@0.8.1 start /home/USER/Bureau/labo/4/ng2-admin-lte
> ng serve --host 0.0.0.0 

** NG Live Development Server is running on http://0.0.0.0:4200 **
 22% building modules 102/158 modules 56 active ...odules/ionicons/dist/css/ionicons.csswebpack: wait until bundle finished: /
Hash: b2b37160c4b548c67211                                                                 
Time: 25236ms
chunk    {0} main.bundle.js, main.bundle.js.map (main) 89.3 kB {3} [initial] [rendered]
chunk    {1} styles.bundle.js, styles.bundle.js.map (styles) 415 kB {4} [initial] [rendered]
chunk    {2} scripts.bundle.js, scripts.bundle.js.map (scripts) 401 kB {4} [initial] [rendered]
chunk    {3} vendor.bundle.js, vendor.bundle.js.map (vendor) 4.89 MB [initial] [rendered]
chunk    {4} inline.bundle.js, inline.bundle.js.map (inline) 0 bytes [entry] [rendered]

ERROR in /home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (2,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEach'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (3,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEachProviders'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (4,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'describe'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (5,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'expect'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (6,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'it'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (9,10): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'ComponentFixture'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (9,28): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'TestComponentBuilder'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (12,36): Cannot find module './app-header.component'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-footer/app-footer.component.spec.ts (38,3): Argument of type '{ directives: any[]; selector: string; template: string; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.

ERROR in /home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-header/app-header.component.spec.ts (2,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEach'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-header/app-header.component.spec.ts (3,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEachProviders'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-header/app-header.component.spec.ts (4,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'describe'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-header/app-header.component.spec.ts (5,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'expect'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-header/app-header.component.spec.ts (6,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'it'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-header/app-header.component.spec.ts (9,10): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'ComponentFixture'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-header/app-header.component.spec.ts (9,28): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'TestComponentBuilder'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/app-header/app-header.component.spec.ts (38,3): Argument of type '{ directives: typeof AppHeaderComponent[]; selector: string; template: string; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.

ERROR in /home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (2,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEach'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (3,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEachProviders'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (4,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'describe'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (5,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'expect'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (6,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'it'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (9,10): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'ComponentFixture'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (9,28): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'TestComponentBuilder'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (12,36): Cannot find module './app-header.component'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/control-sidebar/control-sidebar.component.spec.ts (38,3): Argument of type '{ directives: any[]; selector: string; template: string; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.

ERROR in /home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/menu-aside/menu-aside.component.spec.ts (2,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEach'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/menu-aside/menu-aside.component.spec.ts (3,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEachProviders'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/menu-aside/menu-aside.component.spec.ts (4,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'describe'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/menu-aside/menu-aside.component.spec.ts (5,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'expect'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/menu-aside/menu-aside.component.spec.ts (6,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'it'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/menu-aside/menu-aside.component.spec.ts (9,10): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'ComponentFixture'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/menu-aside/menu-aside.component.spec.ts (9,28): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'TestComponentBuilder'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/menu-aside/menu-aside.component.spec.ts (38,3): Argument of type '{ directives: typeof MenuAsideComponent[]; selector: string; template: string; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.

ERROR in /home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/messages-box/messages-box.component.spec.ts (2,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEach'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/messages-box/messages-box.component.spec.ts (3,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEachProviders'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/messages-box/messages-box.component.spec.ts (4,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'describe'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/messages-box/messages-box.component.spec.ts (5,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'expect'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/messages-box/messages-box.component.spec.ts (6,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'it'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/messages-box/messages-box.component.spec.ts (9,10): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'ComponentFixture'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/messages-box/messages-box.component.spec.ts (9,28): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'TestComponentBuilder'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/messages-box/messages-box.component.spec.ts (38,3): Argument of type '{ directives: typeof MessagesBoxComponent[]; selector: string; template: string; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.

ERROR in /home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/notification-box/notification-box.component.spec.ts (2,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEach'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/notification-box/notification-box.component.spec.ts (3,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEachProviders'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/notification-box/notification-box.component.spec.ts (4,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'describe'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/notification-box/notification-box.component.spec.ts (5,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'expect'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/notification-box/notification-box.component.spec.ts (6,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'it'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/notification-box/notification-box.component.spec.ts (9,10): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'ComponentFixture'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/notification-box/notification-box.component.spec.ts (9,28): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'TestComponentBuilder'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/notification-box/notification-box.component.spec.ts (38,3): Argument of type '{ directives: typeof NotificationBoxComponent[]; selector: string; template: string; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.

ERROR in /home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/tasks-box/tasks-box.component.spec.ts (2,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEach'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/tasks-box/tasks-box.component.spec.ts (3,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEachProviders'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/tasks-box/tasks-box.component.spec.ts (4,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'describe'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/tasks-box/tasks-box.component.spec.ts (5,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'expect'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/tasks-box/tasks-box.component.spec.ts (6,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'it'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/tasks-box/tasks-box.component.spec.ts (9,10): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'ComponentFixture'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/tasks-box/tasks-box.component.spec.ts (9,28): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'TestComponentBuilder'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/tasks-box/tasks-box.component.spec.ts (38,3): Argument of type '{ directives: typeof TasksBoxComponent[]; selector: string; template: string; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.

ERROR in /home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/user-box/user-box.component.spec.ts (2,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEach'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/user-box/user-box.component.spec.ts (3,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'beforeEachProviders'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/user-box/user-box.component.spec.ts (4,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'describe'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/user-box/user-box.component.spec.ts (5,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'expect'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/user-box/user-box.component.spec.ts (6,3): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/core/testing/index"' has no exported member 'it'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/user-box/user-box.component.spec.ts (9,10): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'ComponentFixture'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/user-box/user-box.component.spec.ts (9,28): Module '"/home/USER/Bureau/labo/4/ng2-admin-lte/node_modules/@angular/compiler/testing/index"' has no exported member 'TestComponentBuilder'.
/home/USER/Bureau/labo/4/ng2-admin-lte/src/app/widgets/user-box/user-box.component.spec.ts (38,3): Argument of type '{ directives: typeof UserBoxComponent[]; selector: string; template: string; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.
webpack: Failed to compile.
```

**Comment by mohsin917** [comment]
Hi, I am having the same issue.....

**Comment by TwanoO67** [comment]
The problem is introduced by @angular 2.4.10 going back to 2.4.8 solve.

We should look more in depth in what is the issue with those spec files.
But for the moment I provided the "yarn.lock" file to lock the dependencies.
So you can fixe that by doing a "yarn install" instead of "npm install"

**Comment by TwanoO67** [comment]
Any confirmation ? is it ok for you guys @memento @mohsin917 @peter9208 ?

**Comment by peter9208** [comment]
@TwanoO67 
Thanks for ur help, I solved temporarily by deleting the spec.ts files.

I found something you might be interested in:
- http://stackoverflow.com/questions/38895087/tests-are-failing-after-angular-2-rc5/38895372
- http://stackoverflow.com/questions/38871261/error-angular-core-testing-has-no-exported-member-beforeeach-after-upgradi

I am still looking for what happened with TestComponentBuilder because it isn't in core/testing module anymore. But the beforeEachProvider to TestBed seems like working.
